modules/database/src/std/rec/biRecord.c

```diff
@@ -181,11 +181,9 @@ static long get_enum_str(const DBADDR *paddr, char *pstring)
     if(index!=biRecordVAL) {
         strcpy(pstring,"Illegal_Value");
     } else if(*pfield==0) {
-        strncpy(pstring,prec->znam,sizeof(prec->znam));
-        pstring[sizeof(prec->znam)] = 0;
+        sprintf(pstring, "%.*s", MAX_STRING_SIZE-1, prec->znam);
     } else if(*pfield==1) {
-        strncpy(pstring,prec->onam,sizeof(prec->onam));
-        pstring[sizeof(prec->onam)] = 0;
+        sprintf(pstring, "%.*s", MAX_STRING_SIZE-1, prec->onam);
```

Comment:
What was the complaint here? Replacing a simple byte-copy with string formatting seems a bit heavy-handed to me, although I don't feel like doing performance measurements.

Reply:
Ok, I can see why the compiler is not too happy with `strncpy(pstring,prec->onam,sizeof(prec->onam));`, I am not exactly thrilled either :) . But what about `strncpy(pstring,prec->onam,MAX_STRING_SIZE);`?

In any case, it seems to me what is really needed is `strlcpy()`. Given how many times you've used `snprintf()` only to ensure nul termination, this may be a good opportunity to add it to epics-base. Perhaps under a different name so as not to conflict with BSD.

Reply:
Could be worse is ever called with a buffer `<MAX_STRING_SIZE` but `>=sizeof(prec->onam)`. (Note that `ONAM` is currently in fact shorter.) Be aware that `strncpy` always writes the full buffer, padding it with zeros!
What we really need is a better API with a buffer size argument. Or an API that simply returns a `const char *` and lets the caller copy the data. I think as it is now, `strcpy` is the best option.

Reply:
Huh, you're right. I never paid much attention to string field sizes, and I see now that these are mere 26 chars long :/ . That ... can break expectations.